### ISSUE TITLE
What's the expected behavior of inserting without awaiting runAsync? 

### ISSUE THREAD

**Issue by domoritz** [vega/vega#2513]
I am wondering about the expected behavior when I run something like 

```js
for (const i of [1,2,3,4,5,6,7,8,9,10]) {
  view.insert("foo", [{a: i, b: i}]).runAsync()
}
```

Vega does not process all insertions and therefore some data points are missing. 

<img width="208" alt="Screen Shot 2020-04-06 at 17 50 53" src="https://user-images.githubusercontent.com/589034/78618362-2bb06d00-782f-11ea-8e32-58c81c1912a1.png">

The [docs](https://vega.github.io/vega/docs/api/view/#dataflow-and-rendering) say that I should call `runAsync' after inserting some data but do not specify that I have to await them.

Would it make sense to queue updates here?

Working example at http://bl.ocks.org/domoritz/d5dfd5d195aa7dc838999a25f2dfae02/a82bd83499389296ab6cfcf8987aacbb7eaf5dc4

**Comment by roying** [comment]
The issue may not be specific to _view.insert()_ but to _view.runAsync()_ in general. In Vega View API documentation, _await_ is required when _view.runAsync()_ is used within an _async_ function:
https://vega.github.io/vega/docs/api/view/
 
![image](https://user-images.githubusercontent.com/37225856/78792158-c79cbe80-797e-11ea-83e1-ba5f3ee31b43.png)
Adding _await_ gives the expected result.
```
.then(async (result) => {
      const view = result.view;
      
      for (const i of [1,2,3,4,5,6,7,8,9,10]) {
        await view.insert("foo", [{a: i, b: i}]).runAsync()
      }
```
![image](https://user-images.githubusercontent.com/37225856/78790444-83a8ba00-797c-11ea-8f1b-d37027bb1e39.png)

**Comment by jheer** [comment]
The expected behavior is that one should use await (or Promise chaining) before issuing additional changesets. (As currently implemented, applying a new changeset such as an insert before a previous one has been processed can overwrite the prior changeset.) At minimum, we should update the documentation to reflect this.

**Comment by jheer** [comment]
Fixed in db60923.
